templates/terraform/resource.erb

```diff
@@ -269,7 +269,11 @@ res, err := sendRequest(config, "<%= object.read_verb.to_s.upcase -%>", url, nil
 <%    end -%>
 <%  end -%>
 <%  if object.has_self_link -%>
+<% if version == "ga" -%>
     if err := d.Set("self_link", ConvertSelfLinkToV1(res["selfLink"].(string))); err != nil {
+<% else -%>
+    if err := d.Set("self_link", ConvertSelfLinkToBeta(res["selfLink"].(string))); err != nil {
```

Comment:
Is this going to be a breaking change for all the beta resources that have been using the v1 self link?

Reply:
The pattern has been to convert to V1 even if the resource was created/read using the beta provider so that we don't end up with diffs on resources that interpolate that field. Does something not work without this, or was it just a notice-with-your-eyes thing?

Reply:
Executing a test that creates and references an external VPN gateway within a VPN tunnel was failing. The API is returning the full resource link `beta/projects...` which was diffing with the self_link exported by the external_gateway resource.

Maybe I need a custom expander on the tunnel read action that converts to V1?

Reply:
Ah, you probably want a ResourceRef on the external gateway field of the tunnel resource- the templates should kick in at that point to do all the appropriate URL stuff